setup/symfony_server.rst

```diff
@@ -0,0 +1,271 @@
+Symfony Local Web Server
+========================
+
+You can run Symfony applications with any web server (Apache, nginx, the
+internal PHP web server, etc.) However, Symfony provides its own web server to
+make you more productive while developing your apps.
+
+Although this server is not intended for production use, it supports HTTP/2, TLS/SSL,
+automatic generation of security certificates, local domains, and many other
+features that sooner or later you'll need when developing web projects.
+
+Installation
+------------
+
+The Symfony server is distributed as a free installable binary without any
+dependency and support for Linux, macOS and Windows. Go to `symfony.com/download`_
+and follow the instructions for your operating system.
+
+Getting Started
+---------------
+
+The Symfony web server is started once per project, so you may end up with
+several instances (each of them listening to a different port). This is the
+common workflow to serve a Symfony project:
+
+.. code-block:: terminal
+
+    $ cd my-project/
+    $ symfony server:start
```

Comment:
You must add the `--no-tls` flag here to avoid an error message (TLS no configured yet at this point as you talk about it later on).

Reply:
I don't like that ... but I'll add it if it's really mandatory. What I'd like to have is this:

* `symfony server:start` allows browsing HTTP but not HTTPS (`--no-tls` is not necessary)
* If I try to browse `https://...` I see an informative error telling me that TLS is not enabled yet and that I should run `symfony server:ca:install` to enable it.

If you agree, is it possible to implement this? Thanks!

Reply:
The problem is that you can't serve HTTPS traffic as you have no certificate generated yet.

Reply:
But we can probably turn the error when not using `--no-tls` to a warning instead. WDYT?

Reply:
@tucksaun which one of these is more accurate?

1) Symfony server supports both HTTP and HTPPS. HTTP always works. HTTPS needs some previous work.
2) Symfony server only supports HTTPS. HTTP will never ever work.

I think 1) is true ... that's why this article is designed in a gradual way: 1) Install Server; 2) Use it (with HTTP); 3) Explain HTTPS, install it and use it for the rest of the article.

I don't want to show HTTPS as the first thing (... but I'll do if Symfony server doesn't work with HTTP).

Thanks.

Reply:
It's more nuanced than this:
- Symfony server supports both HTTP and HTTPS.
- HTTP always works but is by default redirected to HTTPS once certificates are configured (one can use `--allow-http` with `symfony server:start` to prevent automatic redirection from HTTP to HTTPS).
- `server:start` *currently* requires the `server:ca:install` command to be run before except if ran with the `no-tls` flag.

We can probably chang the last fact so that `server:start` gets more easy to use for new setups and keep the automatic redirection to HTTPS by default.

Reply:
@javiereguiluz the v4.3.3 version has been released, we now only issue a warning.